go/jsonclient/client.go

```diff
@@ -0,0 +1,191 @@
+package jsonclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	ct "github.com/google/certificate-transparency/go"
+	"golang.org/x/net/context"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+// JSONClient provides common functionality for interacting with a JSON server
+// that uses cryptographic signatures.
+type JSONClient struct {
+	uri        string                // the base URI of the server. e.g. http://ct.googleapis/pilot
+	httpClient *http.Client          // used to interact with the server via HTTP
+	Verifier   *ct.SignatureVerifier // nil if no public key for server available
+}
+
+// New constructs a new JSONClient instance, for the given base URI, using the
+// given http.Client object (if provided).  If a PEM encoded public key is
+// provided, the JSONClient instance will be able to check signatures on signed
+// responses from the server.
+func New(uri string, hc *http.Client, pemEncodedKey string) (*JSONClient, error) {
+	var verifier *ct.SignatureVerifier
+	if pemEncodedKey != "" {
```

Comment:
I've shifted to `pemKey` as the parameter name, and it's now only used on 2 lines.